packages/medusa-plugin-auth/src/auth-strategies/auth0/__tests__/store/verify-callback.spec.ts

```diff
@@ -0,0 +1,161 @@
+	let refreshToken: string;
+	let profile: Profile;
+	let extraParams: ExtraParams;
+	let auth0StoreStrategy: Auth0StoreStrategy;
+
+	beforeEach(() => {
+		profile = {
+			emails: [{ value: existsEmail }],
+		};
+		
+		extraParams = {};
+
+		container = {
+			resolve: <T>(name: string): T => {
+				const container_ = {
+					manager: {
+						transaction: function (cb) {
+							return cb();
+						},
+					},
+					customerService: {
+						withTransaction: function () {
+							return this;
+						},
+						create: jest.fn().mockImplementation(async () => {
+							return { id: 'test' };
+						}),
+            update: jest.fn().mockImplementation(async () => {
+              return {id: 'test'}
+            }),
+						retrieveRegisteredByEmail: jest.fn().mockImplementation(async (email: string) => {
+							if (email === existsEmail) {
+								return {
+									id: 'test',
+								};
+							}
+
+							if (email === existsEmailWithMeta) {
+								return {
+									id: 'test2',
+									metadata: {
+										[CUSTOMER_METADATA_KEY]: true,
+									},
+								};
+							}
+
+							if (email === existsEmailWithMetaAndProviderKey) {
+								return {
+									id: 'test3',
+									metadata: {
+										[CUSTOMER_METADATA_KEY]: true,
+										[AUTH_PROVIDER_KEY]: AUTH0_STORE_STRATEGY_NAME
+									},
+								};
+							}
+
+							if (email === existsEmailWithMetaButWrongProviderKey) {
+								return {
+									id: 'test4',
+									metadata: {
+										[CUSTOMER_METADATA_KEY]: true,
+										[AUTH_PROVIDER_KEY]: 'fake_provider_key'
+									},
+								};
+							}
+
+							return;
+						}),
+					},
+				};
+
+				return container_[name];
+			},
+		} as MedusaContainer;
+
+		auth0StoreStrategy = new Auth0StoreStrategy(
+			container,
+			{} as ConfigModule,
+			{ auth0Domain: 'fakeDomain', clientID: 'fake', clientSecret: 'fake', store: { callbackUrl: '/fakeCallbackUrl'} } as Auth0Options
+		);
+	});
+
+	afterEach(() => {
+		jest.clearAllMocks();
+	});
+
+	it('should succeed', async () => {
+		profile = {
+			emails: [{ value: existsEmailWithMetaAndProviderKey }],
+		};
+
+		const data = await auth0StoreStrategy.validate(req, accessToken, refreshToken, extraParams, profile);
+		expect(data).toEqual(
+			expect.objectContaining({
+				id: 'test3',
+			})
+		);
+	});
+
+	it('should fail when the customer exists without the metadata', async () => {
+		profile = {
+			emails: [{ value: existsEmail }],
+		};
+
+		const err = await auth0StoreStrategy.validate(req, accessToken, refreshToken, extraParams, profile).catch((err) => err);
+		expect(err).toEqual(new Error(`Customer with email ${existsEmail} already exists`));
+	});
+
+	it('should update customer metadata when the customer exsits with ONLY customer metadata key', async () => {
+		profile = {
+			emails: [{ value: existsEmailWithMeta }],
+		};
+
+    const data = await auth0StoreStrategy.validate(req, accessToken, refreshToken, extraParams, profile);
```

Comment:
todo: We should check that the update have been called here. also with the right input data

Reply:
I'm having difficulty mocking the `update` with Jest, I have dm'd you 😇

Reply:
todo: Possibly update the tests for the input data. As of now it is a bit more general in all providers